### ISSUE TITLE
dotnet pack fails with project references and <TargetFramework>

### ISSUE THREAD

**Issue by joelverhagen** [NuGet/Home#3865]
## Steps

1. `dotnet new` two .csproj
1. Add a project reference from one to another (projectA -> projectB)
1. Make sure both have `<TargetFramework>` not `<TargetFrameworks>`
1. `dotnet restore` both
1. `dotnet build` both
1. `dotnet pack` projectA

## Expected

projectA should have a .nupkg produced.

## Actual

```
The "PackTask" task failed unexpectedly.
System.IO.FileNotFoundException: File not found: 'C:\Users\jver\Desktop\repro\packpr\obj\Debug\packpr.dll'.
   at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude)
   at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files)
   at NuGet.Commands.MSBuildProjectFactory.CreateBuilder(String basePath, NuGetVersion version, String suffix, Boolean buildIfNeeded, PackageBuilder builder)
   at NuGet.Commands.PackCommandRunner.BuildFromProjectFile(String path)
   at NuGet.Commands.PackCommandRunner.BuildPackage(String path)
   at NuGet.Commands.PackCommandRunner.BuildPackage()
   at NuGet.Build.Tasks.Pack.PackTask.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__25.MoveNext()
```

**Comment by joelverhagen** [comment]
`dotnet --info`

```
.NET Command Line Tools (1.0.0-preview3-004007)

Product Information:
 Version:            1.0.0-preview3-004007
 Commit SHA-1 hash:  80ec02b4da

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
```

**Comment by zhili1208** [comment]
Looks like it's not our issue.
the issue only happens when project A reference project B.

Msbuild will pass two paths to nuget pack task in TargetPathsToAssemblies like following:
C:\Users\[username]\Documents\test\testproj5\obj\Debug\netstandard1.4\testprojA.dll
C:\Users\[username]\Documents\test\testproj5\obj\Debug\testprojA.dll

the first one is right one, the second one doesn't exist, that's why nuget throw.

those two paths are returned by msbuild task Targets="BuiltProjectOutputGroup;DocumentationProjectOutputGroup"

**Comment by nguerrera** [comment]
See https://github.com/dotnet/sdk/issues/356#issuecomment-258374983. The issue is in the way the msbuild calls in the pack targets are batched and not a problem in the SDK targets.

**Comment by nguerrera** [comment]
When I remove all of the seemingly unnecessary usage of %(_MSBuildProjectReferenceExistent) in the local copy of the pack targets in my package cache, then the pack succeeds in this case.

**Comment by rainersigwald** [comment]
I agree that that usage seems unnecessary--it's creating multiple invocations of the same project for combinations of TargetFrameworks (desired) + various project-reference properties (not needed since they don't apply to the project being built.

**Comment by joelverhagen** [comment]
This has only been checked into dev (4.0.0-rc2). This should be closed when we cherry-pick to 4.0.0-rc branch.
